lib/Dialect/HW/HWOps.cpp

```diff
@@ -1029,6 +1029,26 @@ LogicalResult HWModuleOp::verify() { return verifyModuleCommon(*this); }
 
 LogicalResult HWModuleExternOp::verify() { return verifyModuleCommon(*this); }
 
+void HWModuleOp::addOutputs(
+    SmallVectorImpl<std::pair<StringAttr, Value>> &outputs) {
+  // Rewrite the port list of the module.
+  const unsigned index = getResultTypes().size();
```

Comment:
I'm wondering if it would be useful to provide this index as part of the `outputs` function parameter... That would allow users to additionally take control of the index.

Reply:
I'll rename this `appendOutputs` then, I really want a method that just sticks ports to the end of the list without having to worry about conjuring up an index to the end.